src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax;
+
+public class FigureSupplier extends ColorSupplier {
+    public static final int NUMBER_OF_FIGURES_NAME = 5;
+
+    public FigureSupplier() {
+
+    }
+
+    public String getRandomFigureName() {
+        return FigureName.values()[getRandom().nextInt(NUMBER_OF_FIGURES_NAME)].name();
+    }
+
+    public Figure getRandomFigure() {
+        return new Figure(getRandomColor(), getRandomFigureName());
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle("white", 10, "circle");

```

Comment:
The `getDefaultFigure` method is returning a `Circle` object directly. If `Figure` is intended to be an abstract class or interface, you should not be able to instantiate it directly. Make sure to return a specific figure instance, such as a new `Circle`.